### ISSUE TITLE
Unexpected error when adding multiple strings to a Path using "/" 

### ISSUE THREAD

**Issue by fabiencelier** [microsoft/pyright#147]
**Describe the bug**
When concatenating multiple string to a Path using the "/" operator, Pyright throws the following error : 
> Operator '/' not supported for types 'Literal['XXX']' and 'Literal['YYY']'

**To Reproduce**
```python
from pathlib import Path

p = Path('/etc')
q = p / 'init.d' / 'reboot'
```

**Expected behavior**
It should not be an error as it results in a valid path. It is even in an example of the [pathlib doc](https://docs.python.org/3/library/pathlib.html)

If I wrap it with parenthesis it works ( `(p / 'init.d') / 'reboot'`) but I should not have to do that.

**VS Code extension or command-line**
command line version : pyright 1.0.38
VS code extension version :  pyright 1.0.38

**Comment by erictraut** [comment]
Thanks for the bug report. This was actually a bug in the parser, not the type checker itself. The parser was generating right-handed parse trees for chained binary operators, whereas it should have been generating left-handed parse trees. Because operations were grouped on the right, the type checker tried to evaluate 'init.d' / 'reboot' first, which it identified as an error.

This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.39, which I just published.
